src/async-wrap.cc

```diff
@@ -181,6 +183,33 @@ static void Initialize(Local<Object> target,
 }
 
 
+void AsyncWrap::DestroyIdsCb(uv_idle_t* handle) {
+  Environment* env = Environment::from_destroy_ids_idle_handle(handle);
+  // This doesn't leak a handle.
+  Local<Function> fn = env->async_hooks_destroy_function();
+  int64_t current_id;
+
+  uv_idle_stop(handle);
+
+  if (fn.IsEmpty())
+    return env->erase_destroy_ids();
+
+  while (env->destroy_ids_length() > 0) {
+    current_id = env->pop_destroy_id();
+    HandleScope scope(env->isolate());
+    Local<Value> argv = Number::New(env->isolate(), current_id);
+    TryCatch try_catch(env->isolate());
```

Comment:
You could lift the TryCatch out of the loop, that would be (marginally) more efficient.